src/lib/y2network/sysconfig/config_reader.rb

```diff
@@ -22,44 +22,47 @@
 require "y2network/routing"
 require "y2network/sysconfig_paths"
 require "y2network/routing_table"
-require "y2network/sysconfig_routes_file"
+require "y2network/sysconfig/routes_file"
+require "y2network/sysconfig/dns_reader"
+require "y2network/sysconfig/interfaces_reader"
+require "y2network/interfaces_collection"
 
 Yast.import "NetworkInterfaces"
 
 module Y2Network
-  module ConfigReader
+  module Sysconfig
     # This class reads the current configuration from `/etc/sysconfig` files
-    class Sysconfig
+    class ConfigReader
+      include Yast::Logger
+
       def initialize(_opts = {})
       end
 
       # @return [Y2Network::Config] Network configuration
       def config
-        interfaces = find_interfaces
-        routing_tables = find_routing_tables(interfaces)
+        routing_tables = find_routing_tables(interfaces_reader.interfaces)
```

Comment:
Following the naming patterns, I was surprised not to find a Sysconfig::RoutingReader class. Should it exist? 

Same for Sysconfig::RoutingWriter.